### PR TITLE
Updated code as per Solidified Audit Report

### PR DESCRIPTION
**Pull request by ksridharbabuus**
Updated code as per Solidified Audit Report

### CODE REVIEW

**Comment by raamb** [comment]
@astroseger please review the change.

**Comment by astroseger** [comment]
My biggest concern is following:
As I see you've removed ```settleFundsAndChangeStatus``` (which automatically returned funds back) from ```rejectRequest``` and ```closeRequest```. Which I'm totally fine with, but it was only one advantage of the current algorithm with limited number of Stakers in respect of proposed solution with unlimited number of Stackers (see https://github.com/singnet/snet-rfai/issues/3). 

So, If we don't want to make automatic refunding  in ```rejectRequest``` and ```closeRequest```, then we could simply implement algorithm proposed in #3 and have unlimited number of Stakers.

As I've explained here https://github.com/ksridharbabuus/RFAI-Contracts/issues/2 we can easily mimic the current contract API for end-user (almost...)

**Review by astroseger — APPROVED** [review]
@raamb  @ksridharbabuus All changes in contract seems good to me. So I approve. But as I've said in my previous comment we can do it with unlimited number of Stakers...

File: contracts/ServiceRequest.sol [hunk 1]
```diff
@@ -279,13 +282,10 @@ contract ServiceRequest {
...
         
-        // Should be ative foundation Member
-        require(foundationMembers[msg.sender].status);
-        
         Request storage req = requests[requestId];
-        
-        // Request should be active
-        require(req.status == RequestStatus.Approved);  
+
+        // Should be ative foundation Member or Request Owner
```
Comment (astroseger) [inline contracts/ServiceRequest.sol]: (it seems it was lost in my first "review").
There is a typo in comment :) (ative)

**Comment by ksridharbabuus** [comment]
@astroseger 
The main reason for the removal of function settleFundsAndChangeStatus is to avoid the for loops in settling the funds back and to divide the gas consumption across the staking members by their own settlement rather than spent by the Foundation Member or Request Owner (Either Request Owner or Foundation owner can close the request).

Also I have tried with the unlimited staking and voting and we will ran into the gas limits as the array size increases while calling the function. That is one of the main reason I went through the limited number of stakes based on your suggestion.

**Pull request merged**